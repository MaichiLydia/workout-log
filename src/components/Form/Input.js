import React from 'react'
import { isGreaterDate, timerValidate } from '../../helpers'
import PropTypes from 'prop-types'

function Input(props){

    function inputValidate(e){ 
        const name = e.target.name;
        const value = e.target.value;


        if(props.required && value.trim() === ''){
            props.onChange(name, value, "Required Field")
            return
        }

        if(props.type === 'time' && !(timerValidate(e.target.value, props.min))){
            props.onChange(name, value, e.target.name + ' must be greater than ' + props.min )
            return
        }

        if(props.type === 'date' && !(isGreaterDate(e.target.value, props.min)) ){
            props.onChange(name, value, " Date must be greater than 01 Jan 2000");
            return
        }
        props.onChange(name,value);
    }

    if(props.type === 'date' || props.type === 'time'){
        return (
        <input 
            className="form__input"
            id={props.id}
            type={props.type}
            name={props.name}
            placeholder={props.placeholder}
            min={props.min}
            max={props.max}
            onBlur={inputValidate}
             />

        )       
    }
    return (
       
        <input 
            className="form__input"
            id={props.id}
            type={props.type}
            name={props.name}
            min={props.min}
            max={props.max}
            placeholder={props.placeholder}
            onChange={inputValidate} />
    )
}

Input.propTypes = {
    id: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    required: PropTypes.bool,
    min: PropTypes.string,
    max: PropTypes.string,
    placeholder: PropTypes.string,
    onChange: PropTypes.func

}
export default Input;
