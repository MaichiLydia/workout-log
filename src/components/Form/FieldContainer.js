import React from 'react';
import Input from './Input';
import Select from './Select';
import PropTypes from 'prop-types';


function FieldContainer(props){
    return (
        <div className="form__field-container">
            <label className="form__field-label" htmlFor={props.htmlFor}>
                {props.title}
            </label>
            { props.children}
            
            { props.error && (
                <p className="form__field-container--error">
                    {props.error}
                </p>
            )}
        </div>
    )
}

FieldContainer.Input = Input;
FieldContainer.Select = Select;

FieldContainer.propTypes = {
    htmlFor: PropTypes.string,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ]).isRequired 
}

export default FieldContainer;