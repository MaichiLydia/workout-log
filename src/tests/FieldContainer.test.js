import React from "react";
import FieldContainer from "../components/Form/FieldContainer"
import Input from '../components/Form/Input' 
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';


it('check if FieldContainer receives the right values', () => {
  const props = {
      htmlFor: "name",
      children: <Input id="name"
      type="text"
      name="name"
      required={true}
      onChange={()=> true} />,
      }
  const fieldContainer = mount(<FieldContainer {...props}/>)
  expect(toJson(fieldContainer)).toMatchSnapshot()
});
