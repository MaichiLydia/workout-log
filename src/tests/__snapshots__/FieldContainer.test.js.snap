// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if FieldContainer receives the right values 1`] = `
<FieldContainer
  htmlFor="name"
>
  <div
    className="form__field-container"
  >
    <label
      className="form__field-label"
      htmlFor="name"
    />
    <Input
      id="name"
      name="name"
      onChange={[Function]}
      required={true}
      type="text"
    >
      <input
        className="form__input"
        id="name"
        name="name"
        onChange={[Function]}
        type="text"
      />
    </Input>
  </div>
</FieldContainer>
`;
